hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/OMKeyRequestFactory.java

```diff
@@ -66,43 +55,56 @@ public static OMKeyRequest createRequest(OMRequest omRequest,
     switch (cmdType) {
     case CreateDirectory:
```

Comment:
As per the online discussion with @JyotinderSingh the idea came to remove the OmKeyReqFactory and can replace the check like below:

 ```
   case CreateDirectory:
      keyArgs = omRequest.getCreateDirectoryRequest().getKeyArgs();
      volName = keyArgs.getVolumeName();
      buckName = keyArgs.getBucketName();
      break;
```